### PR TITLE
[TIR] Misc minor updates

### PR DESCRIPTION
**Pull request by MasterJH5574**
This PR adds some minor changes to TIR codebase.

cc @junrushao1994 @Hzfengsy

### CODE REVIEW

File: python/tvm/driver/build_module.py [hunk 1]
```diff
@@ -231,6 +231,8 @@ def build(
     elif isinstance(inputs, PrimFunc):
         input_mod = lower(inputs, name=name)
     elif isinstance(inputs, tvm.IRModule):
+        if name is not None and name != "default_function":
```
Comment (junrushao1994) [inline python/tvm/driver/build_module.py:234]: Let’s skip this - likely it’s a mistake during rebase

Comment (junrushao1994) [reply python/tvm/driver/build_module.py:234]: See https://github.com/apache/tvm/pull/9678

Comment (MasterJH5574) [reply python/tvm/driver/build_module.py]: Thanks for the reminder! Just reverted the change.

**Review by junrushao1994 — APPROVED** [review]

File: src/tir/ir/stmt_functor.cc [hunk 2]
```diff
@@ -700,10 +700,11 @@ Array<Range> Substitute(const Array<Range>& region, const Map<Var, PrimExpr>& vm
 }
 
 void PreOrderVisit(const ObjectRef& stmt_or_expr,
-                   const std::function<bool(const ObjectRef&)>& fvisit) {
+                   const std::function<bool(const ObjectRef&)>& fvisit, bool visit_init_block) {
```
Comment (Hzfengsy) [inline src/tir/ir/stmt_functor.cc:703]: I think it makes no sense to add a parameter in the base class to determine if we visit the init block. Write an additional visitor may be a better choice.

File: include/tvm/tir/stmt.h [hunk 3]
```diff
@@ -1400,6 +1394,54 @@ constexpr const char* meta_schedule_unroll_explicit = "meta_schedule.unroll_expl
...
 
+/*!
+ * \brief Mark that the block need to add predicate for block var bounds during lowering
+ */
+constexpr const char* require_block_var_bound_predicate = "require_bound_predicate";
+
+/*!
+ * \brief Mark that the block should be further rewritten using tensorization.
+ */
+constexpr const char* meta_schedule_auto_tensorize = "meta_schedule.auto_tensorize";
```
Comment (Hzfengsy) [inline include/tvm/tir/stmt.h:1405]: Some annotations are not needed for now

**Review by Hzfengsy — CHANGES_REQUESTED** [review]

File: src/tir/ir/stmt_functor.cc (hunk 2 above)
Comment (MasterJH5574) [reply src/tir/ir/stmt_functor.cc]: Yeah the param has no usage after confirmation. Just removed it.

**Review by Hzfengsy — APPROVED** [review]

**Pull request merged**